### PR TITLE
fix(core): fix partition does not exist error after partition drop

### PR DESCRIPTION
**Pull request by ideoma**
Fixes underlying reason for #2619 e.g. when reader is pooled and a partition is dropped TableReader does not correctly reload the partitions and produces `Partition 'XXX' does not exist in table 'test_table' directory`

### CODE REVIEW

File: core/src/test/java/io/questdb/griffin/AlterTableDropPartitionTest.java [hunk 1]
```diff
@@ -312,10 +382,10 @@ public void testDropPartitionWrongSeparator() throws Exception {
     @Test
     public void testDropPartitionsByDayUsingWhereClause() throws Exception {
         assertMemoryLeak(() -> {
-                    createX("DAY", 720000000);
+            createX("DAY", 720000000);
```
Comment (puzpuzpuz) [inline core/src/test/java/io/questdb/griffin/AlterTableDropPartitionTest.java:385]: nit: this looks like some broken formatting considering the indentation on L390 and below.

**Review by puzpuzpuz — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|7|7|100.00%|

File: core/src/test/java/io/questdb/griffin/AlterTableDropPartitionTest.java (hunk 1 above)
Comment (ideoma) [reply core/src/test/java/io/questdb/griffin/AlterTableDropPartitionTest.java:385]: How to fix it? Is a setting?

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 7 / 7 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/TableReader.java|7|7|100.00%|

File: core/src/test/java/io/questdb/griffin/AlterTableDropPartitionTest.java (hunk 1 above)
Comment (puzpuzpuz) [reply core/src/test/java/io/questdb/griffin/AlterTableDropPartitionTest.java:385]: That's a good question. It seems that IDEA has some bugs around its formatter. For instance, I tried to get rid of the extra trailing newlines in `DataFrameRecordCursor` today and IDEA auto-formatted the file in a totally broken way.

In the worst case, you could try formatting this method manually, but maybe adding/removing newlines may force IDEA to behave as expected.

**Pull request merged**